src/java/org/apache/cassandra/db/compaction/PendingRepairManager.java

```diff
@@ -483,6 +483,14 @@ public Collection<AbstractCompactionTask> createUserDefinedTasks(Collection<SSTa
         return group.entrySet().stream().map(g -> strategies.get(g.getKey()).getUserDefinedTask(g.getValue(), gcBefore)).collect(Collectors.toList());
     }
 
+    public boolean hasPendingRepairSSTable(TimeUUID sessionID, SSTableReader sstable)
```

Comment:
this isn't thread-safe, but could make it sync to make sure it is.... I only added for a test...

Reply:
added `@VisibleForTesting` and `synchronized`... so its now thread safe (with add/remove sstable), but also document that its for tests